sympy/functions/combinatorial/factorials.py

```diff
@@ -1,18 +1,49 @@
 from typing import List
 
-from sympy.core import S, sympify, Dummy, Mod
+from sympy.core import S, sympify, Mod
 from sympy.core.cache import cacheit
-from sympy.core.compatibility import reduce, HAS_GMPY
-from sympy.core.function import Function, ArgumentIndexError
-from sympy.core.logic import fuzzy_and
-from sympy.core.numbers import Integer, pi
+from sympy.core.compatibility import reduce, HAS_GMPY, ordered
+from sympy.core.function import Function, ArgumentIndexError, _mexpand
+from sympy.core.logic import fuzzy_and, fuzzy_or
+from sympy.core.mul import Mul
+from sympy.core.numbers import Float, Integer, pi
 from sympy.core.relational import Eq
+from sympy.core.symbol import Symbol, Dummy
+from sympy.functions.elementary.piecewise import Piecewise
+from sympy.functions.elementary.integers import floor
+from sympy.logic.boolalg import Or, And
 from sympy.ntheory import sieve
 from sympy.polys.polytools import Poly
+from sympy.utilities.iterables import numbered_symbols
 
 from math import sqrt as _sqrt
 
 
+def int_like(x):
+    if x.is_integer is False:
+        if x.is_Float and int(x) == x:
+            return True
+        return False
+    return x.is_integer
```

Comment:
What's the use case of this function?
```suggestion
def int_like(x):
    if x.is_Float and int(x) == x:
        return True
    return x.is_integer
```
x.is_Float is True then x.is_integer should be false.

Reply:
...unless it is Symbol. This function is used to identify when something should be processed as an integer and returns True for `Float(1)`, `Integer(1)` and `Dummy(integer=True)`.

Reply:
`int_like` for cases such as 2.0, 3.0 etc